release.py

```diff
@@ -0,0 +1,420 @@
+
+# Generate RC
+try:
+    int(rc)
+except ValueError:
+    fail("Invalid release candidate number")
+rc_tag = release_version + '-rc' + rc
+
+delete_gitrefs = True # Since we are about to start creating new git refs, enable cleanup function on failure to try to delete them
+cmd("Checking out current development branch", "git checkout -b %s %s" % (release_version, PUSH_REMOTE_NAME + "/" + dev_branch))
+print("Updating version numbers")
+replace("gradle.properties", "version", "version=%s" % release_version)
+replace("tests/kafkatest/__init__.py", "__version__", "__version__ = '%s'" % release_version)
+# Command in explicit list due to messages with spaces
+cmd("Commiting version number updates", ["git", "commit", "-a", "-m", "Bump version to %s" % release_version])
+# Command in explicit list due to messages with spaces
+cmd("Tagging release candidate %s" % rc_tag, ["git", "tag", "-a", rc_tag, "-m", rc_tag])
+rc_githash = cmd_output("git show-ref --hash " + rc_tag)
+cmd("Switching back to dev branch", "git checkout %s" % DEV_BRANCH_NAME)
+
+# Note that we don't use tempfile here because mkdtemp causes problems with sftp and being able to determine the absolute path to a file.
+# Instead we rely on a fixed path and if it
+work_dir = os.path.join(REPO_HOME, ".release_work_dir")
+if os.path.exists(work_dir):
+    fail("A previous attempt at a release left dirty state in the work directory. Clean up %s before proceeding. (This attempt will try to cleanup, simply retrying may be sufficient now...)" % work_dir)
+os.makedirs(work_dir)
+print("Temporary build working director:", work_dir)
+kafka_dir = os.path.join(work_dir, 'kafka')
+cmd("Creating staging area for release artifacts", "mkdir kafka-" + rc_tag, cwd=work_dir)
+artifacts_dir = os.path.join(work_dir, "kafka-" + rc_tag)
+cmd("Cloning clean copy of repo", "git clone %s kafka" % REPO_HOME, cwd=work_dir)
+cmd("Checking out RC tag", "git checkout -b %s %s" % (release_version, rc_tag), cwd=kafka_dir)
+current_year = datetime.datetime.now().year
+cmd("Verifying the correct year in NOTICE", "grep %s NOTICE" % current_year, cwd=kafka_dir)
+
+with open(os.path.join(artifacts_dir, "RELEASE_NOTES.html"), 'w') as f:
+    print("Generating release notes")
+    subprocess.check_call(["./release_notes.py", release_version], stdout=f)
+
+params = { 'release_version': release_version,
+           'rc_tag': rc_tag,
+           'artifacts_dir': artifacts_dir
+           }
+cmd("Creating source archive", "git archive --format tar.gz --prefix kafka-%(release_version)s-src/ -o %(artifacts_dir)s/kafka-%(release_version)s-src.tgz %(rc_tag)s" % params)
+
+cmd("Building artifacts", "gradle", cwd=kafka_dir, env=jdk7_env)
+cmd("Building artifacts", "./gradlew clean releaseTarGzAll aggregatedJavadoc", cwd=kafka_dir, env=jdk7_env)
+# This should be removed with KAFKA-4421
+cmd("Building artifacts for Scala 2.12", "./gradlew releaseTarGz -PscalaVersion=2.12.1", cwd=kafka_dir, env=jdk8_env)
```

Comment:
Trivial fix, but it would be good to say 2.12 instead of 2.12.1 here so that we get the latest patch release. There's one other place in the file that also needs to be updated.